File: FFCG.G4.Movies/FFCG.G4.Movies.Core/Movie.cs
namespace FFCG.G4.Movies.Core
{
    public class Movie
    {
        public int Id { get; }
        public string Name { get; }
        public string Overview { get; set; }
        public decimal Popularity { get; set; }
        public string PosterPath { get; set; }

        public Movie(int id, string name)
        {
            Id = id;
            Name = name;
        }

        public override bool Equals(object o)
        {
            return o is Movie && ((Movie)o).Id == Id;
        }
    }
}

File: FFCG.G7.Movies/src/Core/Movie.ts
export class Movie {
    id: string;
    name: string;
    overview: string;
    popularity: any;
    posterPath: string;
 
    constructor(id: string, name: string, posterPath: string = '') {
        this.id = id;
        this.name = name;
        this.posterPath = posterPath
    }
}
